tests/taglib/test_aiff.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs::File;
+use std::io::{Read, Seek};
+
+use crate::{assert_delta, temp_file};
+use lofty::{Accessor, AudioFile, FileType};
+
+#[test]
+#[ignore]
+fn test_aiff_properties() {
+	let file = lofty::read_from_path("tests/taglib/data/empty.aiff", true).unwrap();
+
+	assert_eq!(file.file_type(), FileType::AIFF);
+
+	let properties = file.properties();
+	assert_eq!(properties.duration().as_secs(), 0);
+	assert_delta!(properties.duration().as_millis(), 67, 1);
+	assert_delta!(properties.audio_bitrate().unwrap(), 706, 1);
+	assert_eq!(properties.sample_rate(), Some(44100));
+	assert_eq!(properties.channels(), Some(1));
+	assert_eq!(properties.bit_depth(), Some(16));
+	// CPPUNIT_ASSERT_EQUAL(2941U, f.audioProperties()->sampleFrames());
+	// CPPUNIT_ASSERT_EQUAL(false, f.audioProperties()->isAiffC());
+}
+
+#[test]
+#[ignore]
+fn test_aifc_properties() {
+	let file = lofty::read_from_path("tests/taglib/data/alaw.aifc", true).unwrap();
+
+	assert_eq!(file.file_type(), FileType::AIFF);
+
+	let properties = file.properties();
+	assert_eq!(properties.duration().as_secs(), 0);
+	assert_delta!(properties.duration().as_millis(), 37, 1);
+	assert_eq!(properties.audio_bitrate(), Some(355));
+	assert_eq!(properties.sample_rate(), Some(44100));
+	assert_eq!(properties.channels(), Some(1));
+	assert_eq!(properties.bit_depth(), Some(16));
+	// CPPUNIT_ASSERT_EQUAL(1622U, f.audioProperties()->sampleFrames());
+	// CPPUNIT_ASSERT_EQUAL(true, f.audioProperties()->isAiffC());
+	// CPPUNIT_ASSERT_EQUAL(ByteVector("ALAW"), f.audioProperties()->compressionType());
+	// CPPUNIT_ASSERT_EQUAL(String("SGI CCITT G.711 A-law"), f.audioProperties()->compressionName());
+}
+
+#[test]
+#[ignore]
+fn test_save_id3v2() {
+	let mut file = temp_file!("tests/taglib/data/empty.aiff");
+
+	{
+		let mut tfile = lofty::read_from(&mut file, true).unwrap();
+
+		assert_eq!(tfile.file_type(), FileType::AIFF);
+
+		assert!(tfile.tag(&lofty::TagType::ID3v2).is_none());
+
+		let mut tag = lofty::Tag::new(lofty::TagType::ID3v2);
+		tag.set_title("TitleXXX".to_string());
+		tfile.insert_tag(tag);
+		file.rewind().unwrap();
+		tfile.save_to(&mut file).unwrap();
+	}
+
+	file.rewind().unwrap();
+
+	{
+		let mut tfile = lofty::read_from(&mut file, true).unwrap();
+
+		assert_eq!(tfile.file_type(), FileType::AIFF);
+
+		let mut tag = tfile.tag(&lofty::TagType::ID3v2).unwrap().to_owned();
+		assert_eq!(tag.title(), Some("TitleXXX"));
+		tag.set_title("".to_string());
+		tfile.insert_tag(tag);
+		file.rewind().unwrap();
+		tfile.save_to(&mut file).unwrap();
+	}
+
+	file.rewind().unwrap();
+
+	{
+		let tfile = lofty::read_from(&mut file, true).unwrap();
+
+		assert_eq!(tfile.file_type(), FileType::AIFF);
+
+		assert!(tfile.tag(&lofty::TagType::ID3v2).is_none());
```

Comment:
Now failing here with:
```
thread 'test_aiff::test_save_id3v2' panicked at 'assertion failed: tfile.tag(&lofty::TagType::ID3v2).is_none()', tests/taglib/test_aiff.rs:86:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/panicking.rs:48:5
   3: taglib::test_aiff::test_save_id3v2
             at ./tests/taglib/test_aiff.rs:86:3
   4: taglib::test_aiff::test_save_id3v2::{{closure}}
             at ./tests/taglib/test_aiff.rs:47:1
   5: core::ops::function::FnOnce::call_once
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/ops/function.rs:248:5
   6: core::ops::function::FnOnce::call_once
             at /rustc/a8314ef7d0ec7b75c336af2c9857bfaf43002bfc/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
test test_aiff::test_save_id3v2 ... FAILED

failures:

failures:
    test_aiff::test_save_id3v2

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 3 filtered out; finished in 0.02s

error: test failed, to rerun pass '-p lofty --test taglib'
```

Reply:
Can we even expect from Lofty that it will remove empty TagType?

Reply:
That's my bad. Missing an early return in `set_title`. You can expect that to work soon.

Reply:
And in general, yes, you can expect an empty tag to be removed. This is just a mistake I made with `Accessor`.

Reply:
Wow, porting only one test, and we find out that many bugs.

Reply:
Verified all tests pass after https://github.com/Serial-ATA/lofty-rs/commit/903531e07f8cbfd2d06d64c07f6541d4e0438872.